corehq/ex-submodules/phonelog/tasks.py

```diff
@@ -17,3 +23,31 @@ def purge_old_device_report_entries():
     UserErrorEntry.objects.filter(server_date__lt=max_age).delete()
     ForceCloseEntry.objects.filter(server_date__lt=max_age).delete()
     UserEntry.objects.filter(server_date__lt=max_age).delete()
+
+
+@task
+def send_device_logs_to_sumologic(domain, xform):
+    url = getattr(settings, 'SUMOLOGIC_URL', None)
+    if url and SUMOLOGIC_LOGS.enabled(xform.form_data.get('device_id'), NAMESPACE_OTHER):
+        for fmt in ['log', 'user_error', 'force_close']:
+            headers = {"X-Sumo-Category": "{env}/{domain}/{fmt}".format(
+                env=_get_sumologic_environment(),
+                domain=domain,
+                fmt=fmt,
+            )}
+            data = getattr(SumoLogicLog(domain, xform), "{}_subreport".format(fmt))()
+            requests.post(url, data=data, headers=headers)
+
+
+def _get_sumologic_environment():
+    """
+    https://docs.google.com/document/d/18sSwv2GRGepOIHthC6lxQAh_aUYgDcTou6w9jL2976o/edit#bookmark=id.ao4j7x5tjvt7  # noqa
+    """
+    if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS:
+        return 'cas'
+    if settings.SERVER_ENVIRONMENT == 'softlayer':
+        return 'india'
+    if settings.SERVER_ENVIRONMENT == 'production':
+        return 'prod'
```

Comment:
@amstone326 is there any reason not to just use the value of `settings.SERVER_ENVIRONEMNT` directly?

Reply:
We can, although the host names will be part of the category that a log is labeled with, and the `settings.SERVER_ENVIRONMENT` value seem somewhat less intuitive than the schema I'm using (softlayer vs. india). Also it looks like there are multiple envs for ICDS; is there some differentiation there that we would want to actually be reflected in the log categorizations? Otherwise those would have to be collapsed into 1 value

Reply:
I wondered this too. @amstone326 I think it might be better to change this setting directly in sumologic, so that if we add new environments we don't need to also add new values here. 

I think the only "confusing" one is "softlayer -> india", but most people already know what this means internally.

Reply:
@proteusvacuum What about ICDS though? Let me know if I'm misunderstanding but it looks like there isn't 1 clear choice for that environment.

Reply:
By host names I assume you mean this environment tag.

Is the concern that people won't know what these refer to? 

Re ICDS there are only 2 because of the migration to a new cluster. The previous one is not live anymore. For future migrations (e.g. prod to AWS) we would probably keep the name the same.

Re India vs SoftLayer, I find 'India' confusing since we have 3 clusters in India now. I agree that 'softlayer' isn't ideal but at least there's no question about what it refers to.

Perhaps the main question is "who is the audience?". If they need to be client facing names then we need to think harder but if it's internal Dimagi then I think creating new names will just add confusion.

Reply:
Hm, your note about what happened when we migrated ICDS does give me reason to worry about doing it this way. If we did for some reason ever change the `SERVER_ENVIRONMENT` value for any environment again, and the person who did so wasn't privy to this particular setup (likely), it would break the log categorization schema because a log cannot be modified in any way once it's ingested by Sumologic. By "break", I mean that we would then have logs from the same environment under 2 different categories, and need to change all of the filters that had previously been searching for logs under the old environment category name to the new one (potentially a huge headache once we have a large number of pre-configured searches, dashboards, user filters, etc. set up in the system).

Re:audience, this is also going to be a mildly user-facing value. 1 of the things that the log category will be used to do is in to set up role-based access control, so that we can give non-Dimagi project admins access to their own logs in Sumologic. So for instance, someone at Tula Salud would be given an account on our Sumologic instance with a Role that filtered for logs with category `prod/tulasalud/*`. We would set this up for them so it doesn't _really_ matter what the value is, but they would see the category labeled on all of their logs. The first half of what I said concerns me much more than this.